test/e2e/network/service.go

```diff
@@ -2679,6 +2681,97 @@ var _ = SIGDescribe("Services", func() {
 
 		framework.ExpectEqual(foundSvc, true, "could not find service 'kubernetes' in service list in all namespaces")
 	})
+
+	ginkgo.It("should test the lifecycle of an Endpoint", func() {
+		ns := f.Namespace.Name
+		testServiceName := "testservice"
+
+		ginkgo.By("creating a service")
+		_, err := f.ClientSet.CoreV1().Services(ns).Create(context.TODO(), &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testServiceName,
+				Namespace: ns,
+				Labels: map[string]string{
+					"testservice-static": "true",
+				},
+			},
+			Spec: v1.ServiceSpec{
+				Type: "ClusterIP",
+				Ports: []v1.ServicePort{{
+					Port:       80,
+					TargetPort: intstr.FromInt(80),
+					Protocol:   v1.ProtocolTCP,
+				}},
+				Selector: map[string]string{
+					"testservice-static": "true",
+				},
+			},
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create service")
+
+		// setup a watch for the Service
+		serviceWatchTimeoutSeconds := int64(60)
+		serviceWatch, err := f.ClientSet.CoreV1().Services(ns).Watch(context.TODO(), metav1.ListOptions{LabelSelector: "testservice-static=true", TimeoutSeconds: &serviceWatchTimeoutSeconds})
+		framework.ExpectNoError(err, "Failed to setup watch on newly created Service")
+		serviceWatchChan := serviceWatch.ResultChan()
+		ginkgo.By("waiting for available Service")
+		for event := range serviceWatchChan {
+			svc, ok := event.Object.(*v1.Service)
+			framework.ExpectEqual(ok, true, "Unable to convert type of Service watch event")
+			if svc.Spec.ClusterIP != "" {
+				break
+			}
+		}
+
+		ginkgo.By("listing all Endpoints")
```

Comment:
you need to create an Endpoints object here. By not setting a selector, we tell the endpoints controller to not do anything with this service. Instead, we manually manage the endpoints (creating them here).

If you do it this way, it will be flaky because you will race in your updates with the EndpointsController.